### PR TITLE
coprocessor: clean up select interface

### PR DESCRIPTION
**Pull request by AndreMouche**
This PR will clean up the select request. You can use dag request instead of the select request. @BusyJay @hicqu @breeswish @ilovesoup @shenli PTAL

### CODE REVIEW

**Comment by hicqu** [comment]
LGTM.

File: src/coprocessor/dag/executor/mod.rs [hunk 1]
```diff
@@ -177,6 +179,7 @@ pub fn build_exec(
                 src,
             )?),
             ExecType::TypeLimit => Box::new(LimitExecutor::new(exec.take_limit(), src)),
+            _ => unimplemented!(),
```
Comment (BusyJay) [inline src/coprocessor/dag/executor/mod.rs]: Why unimplemented?

**Comment by ngaut** [comment]
Always happy to see more clean up.

File: src/coprocessor/dag/executor/mod.rs (hunk 1 above)
Comment (siddontang) [reply src/coprocessor/dag/executor/mod.rs]: @AndreMouche

Comment (AndreMouche) [reply src/coprocessor/dag/executor/mod.rs]: Since other types(TypeStreamAgg..) would be implemented in the later PR.

Comment (siddontang) [reply src/coprocessor/dag/executor/mod.rs]: please add the comment here.

File: src/coprocessor/dag/expr/mod.rs [hunk 2]
```diff
@@ -28,18 +28,70 @@ use std::borrow::Cow;
...
 use coprocessor::codec::mysql::json::JsonDecoder;
 use coprocessor::codec::mysql::{charset, types};
-use coprocessor::codec::Datum;
+use coprocessor::codec::{self, Datum};
 use util;
 use util::codec::number::NumberDecoder;
 use util::codec::Error as CError;
 
-pub use coprocessor::select::xeval::EvalContext as StatementContext;
+/// Flags are used by `SelectRequest.flags` to handle execution mode, like how to handle
```
Comment (breeswish) [inline src/coprocessor/dag/expr/mod.rs]: no longer SelectRequest?

**Comment by breeswish** [comment]
LGTM

**Pull request merged**